contrib/tsdev.sh

```diff
@@ -1,29 +1,79 @@
+	echo
+	echo "Run commands to manage resources on your Timesketch development docker container"
+	echo
+	echo "Options:"
+	echo "  -h, --help        Display this help message"
+	echo
+	echo "Commands:"
+	echo "  build-api-cli     Build and install API and CLI clients"
+	echo "  celery            Start a celery worker"
+	echo "  logs              Follow docker container logs "
+	echo "  shell             Open a shell in the docker container"
+	echo "  test              Execute run_tests.py --coverage"
+	echo "  vue-build         Build the vue frontend"
+	echo "  vue-dev           Serve the vue frontend"
+	echo "  vue-install-deps  Install vue frontend dependencies"
+	echo "  vue-test          Test the vue frontend"
+	echo "  web               Start the web frontend bound to port 5000"
+	echo
+	echo "Examples:"
+	echo "  tsdev.sh logs"
+	echo "  tsdev.sh test"
+	exit 0
+}
 
-if [ $1 == "web" ]; then
-  docker exec -it $CONTAINER_ID gunicorn --reload -b 0.0.0.0:5000 --log-level debug --capture-output --timeout 600 timesketch.wsgi:application
-elif [ $1 == "celery" ]; then
-  docker exec -it $CONTAINER_ID celery -A timesketch.lib.tasks worker --loglevel=info
-elif [ $1 == "vue-install-deps" ]; then
-  docker exec -it $CONTAINER_ID yarn install --cwd=/usr/local/src/timesketch/timesketch/$frontend
-elif [ $1 == "vue-dev" ]; then
-  docker exec -it $CONTAINER_ID yarn run --cwd=/usr/local/src/timesketch/timesketch/$frontend serve
-elif [ $1 == "vue-build" ]; then
-  docker exec -it $CONTAINER_ID yarn run --cwd=/usr/local/src/timesketch/timesketch/$frontend build
-elif [ $1 == "vue-test" ]; then
-  docker exec -it $CONTAINER_ID yarn run --cwd=/usr/local/src/timesketch/timesketch/frontend-ng test
-elif [ $1 == "test" ]; then
-  docker exec -w /usr/local/src/timesketch -it $CONTAINER_ID python3 run_tests.py --coverage
-elif [ $1 == "shell" ]; then
-  docker exec -it $CONTAINER_ID /bin/bash
-elif [ $1 == "logs" ]; then
-  docker logs -f $CONTAINER_ID
-elif [ $1 == "build_api_cli" ]; then
-    docker exec -w '/usr/local/src/timesketch/cli_client/python' -it $CONTAINER_ID python3 setup.py build
-    docker exec -w '/usr/local/src/timesketch/cli_client/python' -it $CONTAINER_ID python3 setup.py install
-    docker exec -w '/usr/local/src/timesketch/api_client/python' -it $CONTAINER_ID python3 setup.py build
-    docker exec -w '/usr/local/src/timesketch/api_client/python' -it $CONTAINER_ID python3 setup.py install
+# Display the help message if there is no command or if the help option is present
+if [ $# -eq 0 ] || [ "$1" == "-h" ]  || [ "$1" == "--help" ]; then
+	help
 fi
+
+# Set variables required to execute commands
+CONTAINER_ID="$(sudo docker container list --filter name='timesketch-dev' --quiet)"
+frontend=${2:-"frontend"}
+
+# Run the provided command
+case "$1" in
+ 	build-api-cli)
+		# CLI client
+		sudo docker exec --workdir '/usr/local/src/timesketch/cli_client/python' --interactive --tty $CONTAINER_ID python3 setup.py build
```

Comment:
I can add a simple check to see if docker needs sudo to run, this can then be used to run the docker commands with/without sudo based a the user's environment.

Reply:
```
# Use "sudo" for docker commands if sudoless docker is not enabled
# "docker version" will return "true" if sudoless docker is configured
if docker version >/dev/null 2>&1; then
        s=""
else
        s="sudo"
        sudo -v
fi

...
...

logs)
                $s docker logs --follow $CONTAINER_ID
```

Reply:
^ This is hacky but faster than actually checking if "rootless" is configured in docker, if it's not it can take a few seconds to fail and time out each time the script runs.

https://sthbrx.github.io/blog/2023/04/05/detecting-rootless-docker/

Reply:
https://github.com/google/timesketch/pull/3081/commits/5d4a76bce3b02f4eff97dc9226ab90ef7f05163d